pandas/tests/frame/test_reductions.py

```diff
@@ -168,7 +168,14 @@ class TestDataFrameAnalytics:
         ],
     )
     def test_stat_op_api_float_string_frame(self, float_string_frame, axis, opname):
-        getattr(float_string_frame, opname)(axis=axis)
+        if opname in ["sum", "min", "max"] and axis == 0:
+            warn = None
+        elif opname not in ["count", "nunique"]:
+            warn = FutureWarning
+        else:
+            warn = None
+        with tm.assert_produces_warning(warn):
```

Comment:
can you add match= here?

are these warnings bc the user needs to change something or bc we need to change something internally?

Reply:
Added match. Concerns numeric_only and nuisance columns, so will have to adjust when removing this